generators/entity-client/files.js

```diff
@@ -18,9 +18,13 @@
  */
 const _ = require('lodash');
 const Randexp = require('randexp');
+const faker = require('faker');
 const utils = require('../utils');
 const constants = require('../generator-constants');
 
+// In order to have consistent results with RandExp, the RNG is seeded.
+Randexp.prototype.randInt = (min, max) => faker.random.number({ min, max });
```

Comment:
It's not used elsewhere in this file, this should be removed.
Further more, I'm not fond of modifying the prototype of something like that. In this case, I see that we set a new function to Randexp:
  - if it's a new one, then it's kinda okay... until Randexp implements a `randIt` function.
  - if a `randInt` already exists and we're overriding it, then what if another developer comes and uses this function not knowing that it has been overridden. You can see the "oops" effect here :)

In my opinion, a better way of doing this would be to extract this function to another file (either new or already existing file) and not modify the prototype so that it may be used.

Reply:
Randexp.prototype.randInt is a function that returns a random value from min to max.

This is a recommended way to do this https://github.com/fent/randexp.js?#custom-prng

Otherwise we must set randInt with a custom value or function every time we call `new Randexp()` or  `const randExp = new this.randexp(fields[idx].fieldValidateRulesPattern)`

Reply:
I can't believe this is actually in their doc. I'm generally okay with monkey-patching, but this is not a good use case.

> Otherwise we must set randInt with a custom value or function every time we call new Randexp() or const randExp = new this.randexp(fields[idx].fieldValidateRulesPattern)

There's an other way:
  - create a RandExp wrapper that will take care of "wrapping" this dependency
  - require this wrapper everywhere where you used to use randexp
That way, you really control how and where this dependency is used.

Reply:
> I can't believe this is actually in their doc. I'm generally okay with monkey-patching, but this is not a good use case.

https://www.npmjs.com/package/randexp#custom-prng

If we use Randexp, for sure we will want to make it reproducible.
In this case I see the "oops" effect when we forget to set randInt and the result is not reproducible.

Reply:
Take a look at:
https://github.com/fent/randexp.js/blob/2b13312e561953288a3cf9b3501e77519b400459/lib/randexp.js#L40-L41

`RandExp.prototype.max` is not set, but RandExp allows to set it and make it default.

Faker too, faker.seed is a global variable.
https://github.com/jhipster/generator-jhipster/blob/4ec3b1f3d216790fdf8e96ad3c0d540012986f7e/generators/entity-server/files.js#L34-L35

An implementation of a local RandExp would be:
```
const creator = new RandExp();
creator.randInt = function ...;
const expr = creator.create(/[0-9]*/).gen();
```
For a consistent result, creator should be passed around, and that doesn't work "out of the box".
That's the reason randomizers uses global implementation.
Creating a randomizer object and passing around seems a waste of time to me.

Faker for instance, there is no easy way to make it an object.

Reply:
Yeah, but there's a difference between:
  -  https://github.com/fent/randexp.js/blob/2b13312e561953288a3cf9b3501e77519b400459/lib/randexp.js#L40-L41 where the lib modifies an intern behavior (everything stays inside the lib here)
  - setting a "global thing", the way the setter works for faker

One way I see it working:
```javascript
const RandExp = require('randexp');

module.exports = { ... };

function randomInteger(min, max) {
  // whatever you want with RandExp
}
function anotherRandomThing(/* whatever args */) {
  // ...
}
```

Why wouldn't it work?

> Faker for instance, there is no easy way to make it an object.

I don't get this.

Reply:
Done.